tests/ga/test_monitor.py

```diff
@@ -546,339 +283,236 @@ def test_reset_loggers_ensuring_timestamp_gets_updated(self, *args):
-
-        # Add agent metric
-        self.event_logger.add_metric("Process", "% Processor Time", "walinuxagent.service", 10)
-
-        # Add agent log
-        self.event_logger.add_log_event(logger.LogLevel.WARNING, "Test sending a log event.")
-
-        # Add extension event file the way extension do it, by dropping a .tld file in the events folder
-        source_file = os.path.join(data_dir, "ext/dsc_event.json")
-        dest_file = os.path.join(conf.get_lib_dir(), "events", "dsc_event.tld")
-        shutil.copyfile(source_file, dest_file)
-
-        # Collect these events and assert they are being sent with the correct sysinfo parameters from the agent
-        with patch.object(protocol, "report_event") as patch_report_event:
-            monitor_handler.collect_and_send_events()
-
-            telemetry_events_list = patch_report_event.call_args_list[0][0][0]
-            self.assertEqual(len(telemetry_events_list.events), 4)
-
-            for event in telemetry_events_list.events:
-                # All sysinfo parameters coming from the agent have to be present in the telemetry event to be emitted
-                for param in monitor_handler.sysinfo:
-                    self.assertTrue(param in event.parameters)
-
-                # The container id, GAVersion are special parameters that are not a part of the static sysinfo parameter
-                # list.
-
-                # The container id value is obtained from the goal state and must be present in all telemetry events.
-                container_id_param = TelemetryEventParam("ContainerId", protocol.client.get_goal_state().container_id)
-                self.assertTrue(container_id_param in event.parameters)
-
-                # Same for GAVersion
-                container_id_param = TelemetryEventParam("GAVersion", CURRENT_AGENT)
-                self.assertTrue(container_id_param in event.parameters)
+        return monitor_handler
+
+    def _create_extension_event(self,
+                               size=0,
+                               name="DummyExtension",
+                               op=WALAEventOperation.Unknown,
+                               is_success=True,
+                               duration=0,
+                               version=CURRENT_VERSION,
+                               message="DummyMessage"):
+        event_data = TestEventMonitoring._get_event_data(name=size if size != 0 else name,
+                op=op,
+                is_success=is_success,
+                duration=duration,
+                version=version,
+                message=random_generator(size) if size != 0 else message)
+        event_file = os.path.join(self.event_dir, "{0}.tld".format(int(time.time() * 1000000)))
+        with open(event_file, 'wb+') as fd:
+            fd.write(event_data.encode('utf-8'))
+
+    @staticmethod
+    def _get_event_data(duration, is_success, message, name, op, version, eventId=1):
```

Comment:
PyCharm identified that this method is nearly identical to `tests.protocol.test_wire.get_event`. In fact, the one in `test_wire` is outdated since we are also adding the `IsInternal` parameter which is removed with this change. Can you please merge the two methods?

Reply:
I had to deal with so many test breaks that I did not cleanup all the tests and some of them (like this one) I only modified to allow them to pass. The original code was creating a weird event that did not follow the schema for an extension or an agent event, but was saving it using the "*.tld" extension so it looked to the new code as an extension event, causing many of these tests to break. Had it been done using the public interface (add_event, for example) the internal refactoring would have not caused any issues with it.

As we do improvements on the existing code we are going to have to decide at what point to stop. I had already done a fair amount of work investigating test failures and rewriting many of them and I did not go further on this one on specific. Each one of us will have to make that call. I stopped here.

When writing new code though, we should make every effort to avoid this kind of issues, though. I believe these tests are relatively recent, but to be fair we have not been paying enough attention to design as a team. Let's try to improve from now on.